src/clover/analysis.py

```diff
@@ -0,0 +1,2852 @@
+            total_electric_load[0:CUT_OFF_TIME][DemandType.COMMERCIAL.value],
+            label=DemandType.COMMERCIAL.value,
+        )
+        plt.plot(
+            range(CUT_OFF_TIME),
+            total_electric_load[0:CUT_OFF_TIME][DemandType.PUBLIC.value],
+            label=DemandType.PUBLIC.value,
+        )
+        plt.plot(
+            range(CUT_OFF_TIME),
+            np.sum(total_electric_load[0:CUT_OFF_TIME], axis=1),
+            "--",
+            label="total",
+        )
+        plt.legend(loc="upper right")
+        plt.xticks(
+            [entry for entry in (range(0, CUT_OFF_TIME - 1, min(4, CUT_OFF_TIME - 1)))]
+        )
+        plt.xlabel("Hour of simulation")
+        plt.ylabel("Electric power demand / kW")
+        plt.title(f"Load profile of the community for the first {CUT_OFF_TIME} hours")
+        plt.savefig(
+            os.path.join(figures_directory, "electric_demands.png"),
+            transparent=True,
+        )
+        plt.close()
+        pbar.update(1)
+
+        # Plot the annual variation of the electricity demand.
+        _, axis = plt.subplots(1, 2, figsize=(8, 4))
+        domestic_demand = np.sum(
+            np.reshape(
+                total_electric_load[0:HOURS_PER_YEAR][DemandType.DOMESTIC.value].values,
+                (365, 24),
+            ),
+            axis=1,
+        )
+        commercial_demand = np.sum(
+            np.reshape(
+                total_electric_load[0:HOURS_PER_YEAR][
+                    DemandType.COMMERCIAL.value
+                ].values,
+                (365, 24),
+            ),
+            axis=1,
+        )
+        public_demand = np.sum(
+            np.reshape(
+                total_electric_load[0:HOURS_PER_YEAR][DemandType.PUBLIC.value].values,
+                (365, 24),
+            ),
+            axis=1,
+        )
+        total_demand = np.sum(
+            np.reshape(
+                np.sum(total_electric_load[0:HOURS_PER_YEAR].values, axis=1),
+                (365, 24),
+            ),
+            axis=1,
+        )
+        axis[0].plot(
+            range(365),
+            pd.DataFrame(domestic_demand).rolling(5).mean(),
+            label="Domestic",
+            color="blue",
+        )
+        axis[0].plot(
+            range(365),
+            pd.DataFrame(commercial_demand).rolling(5).mean(),
+            label="Commercial",
+            color="orange",
+        )
+        axis[0].plot(
+            range(365),
+            pd.DataFrame(public_demand).rolling(5).mean(),
+            label="Public",
+            color="green",
+        )
+        axis[0].plot(range(365), domestic_demand, alpha=0.5, color="blue")
+        axis[0].plot(range(365), commercial_demand, alpha=0.5, color="orange")
+        axis[0].plot(range(365), public_demand, alpha=0.5, color="green")
+        axis[0].legend(loc="best")
+        axis[0].set(
+            xticks=(range(0, 366, 60)),
+            yticks=range(0, 26, 5),
```

Comment:
Here (and other places) we should remove the hard-coding of the axis limits (etc.). I think for now we should accept the files as-is, but add this to the to-do list.